### ISSUE TITLE
Linker prefix configuration ignored

### ISSUE THREAD

**Issue by serializingme** [maven-nar/nar-maven-plugin#332]
It seems the linker prefix property in the `aol.properties` file is ignored. NAR still invokes the normal `g++` linker instead of the prefixed version `<prefix>-g++`.

I have the following NAR configuration in the `pom.xml`:

```
<plugin>
    <groupId>com.github.maven-nar</groupId>
    <artifactId>nar-maven-plugin</artifactId>
    <version>3.6.0</version>
    <configuration>
        <aol>amd64.Windows.mingw</aol>
    </configuration>
</plugin>
```

In the `aol.properties` I have the following:

```
amd64.Windows.linker=mingw

amd64.Windows.mingw.cpp.compiler=g++
amd64.Windows.mingw.cpp.prefix=x86_64-w64-mingw32-
amd64.Windows.mingw.cpp.defines=WIN32_LEAN_AND_MEAN
amd64.Windows.mingw.cpp.options=-Wall
amd64.Windows.mingw.cpp.includes=**/*.cc **/*.cpp **/*.cxx
amd64.Windows.mingw.cpp.excludes=

amd64.Windows.mingw.c.compiler=gcc
amd64.Windows.mingw.c.prefix=x86_64-w64-mingw32-
amd64.Windows.mingw.c.defines=WIN32_LEAN_AND_MEAN
amd64.Windows.mingw.c.options=-Wall
amd64.Windows.mingw.c.includes=**/*.c
amd64.Windows.mingw.c.excludes=

amd64.Windows.mingw.res.compiler=windres
amd64.Windows.mingw.res.prefix=x86_64-w64-mingw32-
amd64.Windows.mingw.res.defines=
amd64.Windows.mingw.res.options=-O coff
amd64.Windows.mingw.res.includes=**/*.rc
amd64.Windows.mingw.res.excludes=

amd64.Windows.mingw.linker.prefix=x86_64-w64-mingw32-
amd64.Windows.mingw.linker.options=-s

amd64.Windows.mingw.lib.prefix=lib
amd64.Windows.mingw.shared.prefix=
amd64.Windows.mingw.static.extension=a
amd64.Windows.mingw.shared.extension=dll
amd64.Windows.mingw.plugin.extension=dll
amd64.Windows.mingw.jni.extension=dll
amd64.Windows.mingw.executable.extension=exe
```

And the execution (`mvn clean install -e -X`) looks like:

```
......
[INFO] Starting Core 0 with 1 source files...
[DEBUG] x86_64-w64-mingw32-windres -O coff -DNDEBUG -I/mnt/hgfs/test/src/main/include -o/mnt/hgfs/test/target/nar/obj/amd64.Windows.mingw/resources.8891d2ad.o res/resources.rc
[DEBUG] Execute:Java13CommandLauncher: Executing 'x86_64-w64-mingw32-windres' with arguments:
'-O'
'coff'
'-DNDEBUG'
'-I/mnt/hgfs/test/src/main/include'
'-o/mnt/hgfs/test/target/nar/obj/amd64.Windows.mingw/resources.8891d2ad.o'
'res/resources.rc'

The ' characters around the executable and arguments are
not part of the command.
[INFO] Found 2 processors available
[INFO] Limited used processors to 1
[INFO] 
Starting Core 0 with 1 source files...
[DEBUG] x86_64-w64-mingw32-gcc -Wall -c -mconsole -DWIN32_LEAN_AND_MEAN -I/mnt/hgfs/test/src/main/include -o /mnt/hgfs/test/target/nar/obj/amd64.Windows.mingw/main.11c3479e.o c/main.c
[DEBUG] Execute:Java13CommandLauncher: Executing 'x86_64-w64-mingw32-gcc' with arguments:
'-Wall'
'-c'
'-mconsole'
'-DWIN32_LEAN_AND_MEAN'
'-I/mnt/hgfs/test/src/main/include'
'-o'
'/mnt/hgfs/test/target/nar/obj/amd64.Windows.mingw/main.11c3479e.o'
'c/main.c'

The ' characters around the executable and arguments are
not part of the command.
[INFO] Linking...
[INFO] Starting link {8 -s -static -static-libgcc /usr/lib/gcc/x86_64-linux-gnu/8/libstdc++.a}
[DEBUG] g++ -s -static -o madeliefje /mnt/hgfs/test/target/nar/obj/amd64.Windows.mingw/main.11c3479e.o /mnt/hgfs/test/target/nar/obj/amd64.Windows.mingw/resources.8891d2ad.o -static-libgcc /usr/lib/gcc/x86_64-linux-gnu/8/libstdc++.a
[DEBUG] Execute:Java13CommandLauncher: Executing 'g++' with arguments:
'-s'
'-static'
'-o'
'madeliefje'
'/mnt/hgfs/test/target/nar/obj/amd64.Windows.mingw/main.11c3479e.o'
'/mnt/hgfs/test/target/nar/obj/amd64.Windows.mingw/resources.8891d2ad.o'
'-static-libgcc'
'/usr/lib/gcc/x86_64-linux-gnu/8/libstdc++.a'

The ' characters around the executable and arguments are
not part of the command.
[ERROR] /usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/8/../../../x86_64-linux-gnu/crt1.o: in function `_start':
[ERROR] (.text+0x20): undefined reference to `main'
[ERROR] collect2: error: ld returned 1 exit status
```

The linker fails because it is using the system compiler, instead of the MinGW cross compiler (hence the prefix `x86_64-w64-mingw32-`. What am I missing?

**Comment by bchiodo** [comment]
The linker prefix wasn't working because the NAR code never consulted the aol.properties file to determine if the linker.prefix option was specified there. Hence, only when setting the prefix via the pom file with `<prefix>my-prefix</prefix>` under the `<linker>` tag would the prefix be honored.

I do not know if this was the intention of the original author of the linker prefix mechanism, or if it was a bug. I have assumed it is a bug since the similar compiler.prefix mechanism _did_ consult the aol.properties file.

I have fixed this bug with 9716e99720b6b9445978d33c00bd50407df14264.

**Comment by serializingme** [comment]
Cheers!
